### DISCUSSION TITLE
Convent XML to DOCX

### DISCUSSION THREAD

**Discussion by mstudioIL** [jgm/pandoc#10136] [Q&A] [answered]
I asked ChatGPT how to convent XML I export from WordPress to DOCX so people can review the text and it offer your tool. I run it and got the message 
"[WARNING] Could not deduce format from file extension .xml
  Defaulting to markdown"
  I used this the "pandoc WordPress.2024-09-01.xml -o WordPress.2024-09-01.docx" I run it on the file folder.
How do I fix this?

**Comment by denismaier** [comment]
The extension xml is generic so you'll have to provide more details about the format used. For example, if you use jats you'd say -f jats.
Don't know if WordPress XML is supported though.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
From: mstudio ***@***.***>
Sent: Sunday, September 1, 2024 11:15:36 AM
To: jgm/pandoc ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jgm/pandoc] Convent XML to DOCX (Discussion #10136)


I asked ChatGPT how to convent XML I export from WordPress to DOCX so people can review the text and it offer your tool. I run it and got the message
"[WARNING] Could not deduce format from file extension .xml
Defaulting to markdown"
I used this the "pandoc WordPress.2024-09-01.xml -o WordPress.2024-09-01.docx" I run it on the file folder.
How do I fix this?

**Reply by mstudioIL** [reply]
Thanks for the answer, what details can I give you, so you can help me better?

**Comment by denismaier** [comment]
Not me. I meant this: pandoc usually can infer the format from the file ending. But in this case you need to specify the format explicitly since XML can be many things.


Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by mstudioIL** [reply]
OK, how do I know want is the format? 
This is the start of the file
```
<?xml version="1.0" encoding="UTF-8" ?>
<!-- This is a WordPress eXtended RSS file generated by WordPress as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the "WordPress" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

	<!-- generator="WordPress/6.5.5" created="2024-09-01 08:48" -->
<rss version="2.0"
	xmlns:excerpt="http://wordpress.org/export/1.2/excerpt/"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:wfw="http://wellformedweb.org/CommentAPI/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:wp="http://wordpress.org/export/1.2/"
>
```
Thanks for the help

**Reply by fiapps** [reply]
That's an RSS feed, apparently with additional nonstandard fields. This is not a format pandoc reads, and is not really even a single document but a collection of documents, each with its own metadata. If you extract the HTML for a single post or page, pandoc can parse that. You could also give pandoc the URL for a post or page and it will download the HTML and convert it. Note that pandoc respects structure but not CSS styles.

**Reply by mstudioIL** [reply]
Thanks for the answer, where I can find example how to use the URL to download the HTML? after I download it will I be able to convert it do DOCX?

**Reply by fiapps** [reply]
Basic usage is just something like this: `pandoc https://wordpress.com/blog/2024/02/27/more-control-over-the-content-you-share/ -o more-control.docx` That will convert all the HTML from the webpage, so you'll get things like webpage headers and footers (but without the styling) in the output file. You'll then have to delete those with a word processor.

Also, when i run this on my computer, I get warnings that `rsvg-convert` is not installed. I just ignored those warnings because I don't think the SVG files it can't convert without that program are part of the blog post. If you do need SVG files converted, you'll have to install that in addition to pandoc.

**Comment by mstudioIL** [comment]
Thanks, I also got this error , but I got the images on the doc, how do I install the addon? do I need it?
Where I read more about the parameters? About the headers and footers can I remove them? they using the `<header></header>` and `<footer></footer>` tags so everything on this tags can be removed.

**Reply by fiapps** [reply, marked as answer]
I don't have a lot of time to help you, but I can add the following.

I have never needed `rsvg-convert` and I can't find any mention of it in the pandoc manual, so I don't know what function it serves. I would say that if the output doesn't seem to be missing any of the images from the article, you shouldn't worry about it. If you do want to install it, see the [pandoc installation instructions](https://pandoc.org/installing.html) for your platform.

A quick experiment shows me that pandoc doesn't represent `<header></header>` and `<footer></footer>` tags in its AST, so if you want to use those, you'll have preprocess the HTML with something that can work on the DOM before you pass it to pandoc. Alternatively, you can look at the HTML and see if what you want to extract is wrapped in a div with a class. Pandoc does keep divs and classes, so this kind of processing can be done in a Lua filter. In the case of the URL I used as an example, the post is wrapped in a div with a "post" class, and there is only one div with this class. So we can use a Lua filter that finds the div with a post class and makes that the whole document. If `extract-blog.lua` contains the following script:

```lua
function Pandoc(doc)
    local post = nil
    doc:walk{
        Div = function (el)
            if el.classes:includes("post") then
                post = el
            end
        end
    }
    if post == nil then
        print("Post not found: keeping whole document.")
    else
        print("Extracted post.")
        doc.blocks = {post}
        return doc
    end
end
```

Then `pandoc https://wordpress.com/blog/2024/02/27/more-control-over-the-content-you-share/ -L extract-blog.lua -o more-control.docx` will create an MS Word document containing only the contents of the div with the post class.

**Reply by mstudioIL** [reply]
Thanks for all the help. I will check what you show. 
I sure the .lua file need to be where I run the command to save the doc.

**Reply by jgm** [reply]
Incidentally, I have long thought that it would be helpful if the HTML reader ignored header, footer, and nav elements  -- or perhaps, alternatively, only converted the main element if one is present.  Perhaps there should be an issue for this?  Anyway, comments welcome.
https://github.com/jgm/pandoc/issues/10140
